core/domain/acl_decorators.py

```diff
@@ -853,6 +854,107 @@ def test_can_edit(self, topic_id, **kwargs):
     return test_can_edit
 
 
+def can_edit_question(handler):
+    """Decorator to check whether the user can edit given question."""
+    def test_can_edit(self, question_id, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        if question_services.check_can_edit_question(
+                self.user.user_id, question_id):
```

Comment:
This looks weird. Should it be self.user_id?

If so, why was this not caught during tests? Please add backend tests to cover potential errors here and elsewhere.